operator/controllers/loki/controllers.go

```diff
@@ -0,0 +1,53 @@
+package controllers
+
+import (
+	"github.com/go-logr/logr"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	lokiv1 "github.com/grafana/loki/operator/apis/loki/v1"
+)
+
+const (
+	AlertingRuleCtrlName        = "alertingrule"
+	CertRotationCtrlName        = "certrotation"
+	DashboardsCtrlName          = "dashboard"
+	LokiStackCtrlName           = "lokistack"
+	LokiStackZoneLabelsCtrlName = "lokistack-zone-labels"
+	RecordingRuleCtrlName       = "recordingrule"
+	RulerConfigCtrlName         = "rulerconfig"
+)
+
+type LogContructorType func(request *reconcile.Request) logr.Logger
```

Comment:
Small typo and suggestion for renaming (don't know if we did a similar thing already):

```suggestion
type LogConstructorFunc func(request *reconcile.Request) logr.Logger
```

---

Also this type does not need to be public, as far as I can see...